lib/mapper.js

```diff
@@ -112,14 +113,30 @@ function common(facade){
   if (facade.city()) ret.city = facade.city();
   if (facade.region()) ret.region = facade.region();
 
-  if (os && os.toLowerCase() == 'ios') {
+  function mobileDeviceType(){
+    if (!library && !os) return 'unknown';
+    if (contains('ios')) return 'ios';
+    if (contains('android')) return 'android';
+    return 'unknown';
+
+    function contains(str) {
+      return getField(library).toLowerCase().indexOf(str) !== -1
+          || getField(os).toLowerCase().indexOf(str) !== -1;
+    }
+
+    function getField(field) {
+      return typeof field === 'string' ? field : field.name;
+    }
```

Comment:
Same with both of these, except they're being reconstructed twice per invocation of `mobileDeviceType`